Source/JavaScriptCore/parser/Lexer.cpp

```diff
@@ -990,10 +990,17 @@ template <bool shouldCreateIdentifier> ALWAYS_INLINE JSTokenType Lexer<LChar>::p
     if (shouldCreateIdentifier || m_parsingBuiltinFunction) {
         std::span identifierSpan { identifierStart, static_cast<size_t>(currentSourcePtr() - identifierStart) };
         if (m_parsingBuiltinFunction && isBuiltinName) {
-            if (isWellKnownSymbol)
+            if (isWellKnownSymbol) {
+#if ASSERT_ENABLED
+                ASSERT_WITH_MESSAGE(m_vm.propertyNames->builtinNames().lookUpWellKnownSymbol(identifierSpan), "Private well-known symbol @@%s not found", StringView(identifierSpan).utf8().data());
+#endif
                 ident = &m_arena->makeIdentifier(m_vm, m_vm.propertyNames->builtinNames().lookUpWellKnownSymbol(identifierSpan));
```

Comment:
Should not need `#if ASSERT_ENABLED`, since `ASSERT_WITH_MESSAGE` expands to nothing if it’s not enabled.

But also, do we really want to call `lookUpWellKnownSymbol` twice? How about we we write this with a local variable instead.

```suggestion
                auto* symbol = m_vm.propertyNames->builtinNames().lookUpWellKnownSymbol(identifierSpan);
                ASSERT_WITH_MESSAGE(symbol, "Private well-known symbol @@%s not found", StringView(identifierSpan).utf8().data());
                ident = &m_arena->makeIdentifier(m_vm, symbol);
```

Reply:
Thanks, updated!